### ISSUE TITLE
Deadlock between ProvSSLSessionContext.removeSession and ProvSSLSessionBase.isValid

### ISSUE THREAD

**Issue by tijoytom-okta** [bcgit/bc-java#1344]
Version: bc-fips:jar:1.0.2.3

```
"http-nio-8443-exec-1":
  waiting to lock monitor 0x00007f89d44e7200 (object 0x0000000700ecbd40, a org.bouncycastle.jsse.provider.ProvSSLSessionContext),
  which is held by "http-nio-8443-exec-86"
"http-nio-8443-exec-86":
  waiting to lock monitor 0x00007f896c178a00 (object 0x000000076a61e4d8, a org.bouncycastle.jsse.provider.ProvSSLSession),
  which is held by "http-nio-8443-exec-28"
"http-nio-8443-exec-28":
  waiting to lock monitor 0x00007f89d44e7200 (object 0x0000000700ecbd40, a org.bouncycastle.jsse.provider.ProvSSLSessionContext),
  which is held by "http-nio-8443-exec-86"
```
[thread.txt](https://github.com/bcgit/bc-java/files/10886787/thread.txt)

I think https://github.com/bcgit/bc-java/commit/571ae2f40c36644d53148e875c23aa540b6277a1 would fix it since it's removing the instance level synchronization. @peterdettman

**Comment by peterdettman** [comment]
What version of bctls-fips jar are you using? Please update to the latest (currently 1.0.14.1).

**Comment by tijoytom-okta** [comment]
@peterdettman  thanks. We are using bctls-fips:jar:1.0.12. I guess updating to 1.0.14.1 should fix this?

**Comment by peterdettman** [comment]
Yes, I believe it was actually fixed in 1.0.12.2.
